codalab/lib/codalab_manager.py

```diff
@@ -49,6 +49,16 @@
 from codalab.model.worker_model import WorkerModel
 
 
+def exception_handler(exctype, value, traceback, debug_hook=sys.excepthook):
```

Comment:
Why put this in `codalab_manager.py`?

Reply:
Also, this seems like a potentially nice way to handle exceptions, but I wonder if it might be confusing / dangerous, since it has a global effect on the entire codebase?  Also, since we don't handle our other error messages through this mechanism...

Reply:
This is not a global change. I put it here because we only want to handle the PermissionError in codalab_manager. I tested that PermissionError raised other places will be displayed in the conventional way.

Reply:
I think it might be nicer to instead just catch PermissionError in cl.py, just like what we do with `KeyboardInterrupt`: https://github.com/codalab/codalab-worksheets/blob/5b83914cf4e73d70a9fe39ce62513c13222c9e1d/codalab/bin/cl.py#L7-L13

Reply:
> I think it might be nicer to instead just catch PermissionError in cl.py, just like what we do with `KeyboardInterrupt`:
> 
> https://github.com/codalab/codalab-worksheets/blob/5b83914cf4e73d70a9fe39ce62513c13222c9e1d/codalab/bin/cl.py#L7-L13

Do we want to handle all PermissionErrors in this way? I noticed that we have multiple places raising PermissionError.

Reply:
Hmm, in that case maybe it would be best to create a subclass of PermissionError called `LoginPermissionError` and then only catch `LoginPermissionError`.

Reply:
The stacktrace doesn't go thru cl.py:
```
Traceback (most recent call last):
  File "/Users/yibol/codalab-worksheets/venv/bin/cl-worker-manager", line 33, in <module>
    sys.exit(load_entry_point('codalab', 'console_scripts', 'cl-worker-manager')())
  File "/Users/yibol/codalab-worksheets/codalab/worker_manager/main.py", line 121, in main
    manager.run_loop()
  File "/Users/yibol/codalab-worksheets/codalab/worker_manager/worker_manager.py", line 125, in run_loop
    self.run_one_iteration()
  File "/Users/yibol/codalab-worksheets/codalab/worker_manager/worker_manager.py", line 161, in run_one_iteration
    'bundles', params={'worksheet': None, 'keywords': keywords, 'include': ['owner']}
  File "/Users/yibol/codalab-worksheets/codalab/client/json_api_client.py", line 18, in wrapper
    return f(*args, **kwargs)
  File "/Users/yibol/codalab-worksheets/codalab/client/json_api_client.py", line 417, in fetch
    query_params=self._pack_params(params),
  File "/Users/yibol/codalab-worksheets/codalab/worker/rest_client.py", line 66, in _make_request
    access_token = self._get_access_token()
  File "/Users/yibol/codalab-worksheets/codalab/lib/codalab_manager.py", line 432, in <lambda>
    lambda: self._authenticate(address, auth_handler),
  File "/Users/yibol/codalab-worksheets/codalab/lib/codalab_manager.py", line 504, in _authenticate
    raise LoginPermissionError("Invalid username or password.")
codalab.common.LoginPermissionError: Invalid username or password.
```
Instead I did this in `codalab-worksheets/codalab/worker_manager/main.py`

Reply:
Screenshot updated.